ui/radar/radar.js

```diff
@@ -0,0 +1,235 @@
+        m['dom'] = tr;
+        this.targetMonsters[mob_name] = m;
+        if (options.TTS) {
+          callOverlayHandler({
+            call: 'cactbotSay',
+            text: m['rank'] + ' ' + m['name'],
+          });
+        } else if (options.PopSoundAlert && options.PopSound && options.PopVolume) {
+          let audio = new Audio(options.PopSound);
+          audio.volume = options.PopVolume;
+          audio.play();
+        }
+      }
+    }
+  }
+
+  UpdateMonsterPuller(log, monster) {
+    let matches = log.match(Regexes.abilityFull({}));
+    if ((monster['puller'] === null))
+      monster['puller'] = matches.groups.source;
+  }
+
+  UpdateMonsterDom(e, monster) {
+    let options = this.options;
+    if (monster['rank'] in options.RankOptions)
+      options = Object.assign({}, this.options, options.RankOptions[monster['rank']]);
+    let tr = monster['dom'];
+    // calculate rotation based on facing
+    let playerVector = new Point2D(e.detail.pos.x, -e.detail.pos.y);
+    let targetVector = monster['pos'];
+    let deltaVector = new Point2D(targetVector.x - playerVector.x, targetVector.y - playerVector.y);
+    if (tr) {
+      tr.childNodes[1].innerHTML = monster['rank'] + '&nbsp;&nbsp;&nbsp;&nbsp;' + monster['name'];
+      if (Math.abs(e.detail.pos.z - monster['pos_z']) > 5)
+        tr.childNodes[1].innerHTML += '&nbsp;&nbsp;' + (e.detail.pos.z < monster['pos_z']? '↑' : '↓');
+      tr.childNodes[1].innerHTML += '<br>' + length(deltaVector).toFixed(2) + 'm';
+      if (Date.now() / 1000 <= monster['battle_time'] + 60) {
+        tr.childNodes[1].innerHTML += ' ' + (monster['current_hp'] * 100 /
+          monster['hp']).toFixed(2) + '%';
+      }
+      if (options.Puller) {
+        if (monster['puller'])
+          tr.childNodes[1].innerHTML += '&nbsp;&nbsp;' + monster['puller'];
+      }
+    }
+    if (options.DetectionRange > 0 && length(deltaVector) > options.DetectionRange)
+      monster['dom'].setAttribute('class', 'hide');
+    else
+      monster['dom'].setAttribute('class', '');
+    let deltaTheta = Math.acos(theta(deltaVector, new Point2D(1, 0)));
+    if (deltaVector.y < 0)
+      deltaTheta = -deltaTheta;
+    deltaTheta += Math.PI - e.detail.rotation;
+    let angle = -deltaTheta*180/Math.PI;
+    let arrow_id = 'arrow-' + monster['id'];
+    let arrow = document.getElementById(arrow_id);
+    arrow.style.transform='rotate('+angle+'deg)';
+  }
+
+  RemoveMonster(mob_name) {
+    if (mob_name in this.targetMonsters) {
+      this.targetMonsters[mob_name]['dom'].remove();
+      delete this.targetMonsters[mob_name];
+    }
+  }
+
+  OnLogEvent(e) {
+    let lang = this.options.Language;
+    for (let i = 0; i < e.detail.logs.length; i++) {
+      // add new combatant
+      let matches = e.detail.logs[i].match(Regexes.addedCombatantFull({}));
+      if (matches) {
+        let monster = this.nameToMonster[matches.groups.name];
+        if (monster)
+          this.AddMonster(e.detail.logs[i], monster);
```

Comment:
Here and in other function calls below like UpdateMonsterPuller, it seems like you could pass matches.groups to the function rather than passing the log and rematching.

Reply:
removed rematching